sklearn/linear_model/cd_fast.pyx

```diff
@@ -201,26 +240,25 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 
                 if w_ii != 0.0:
                     # R += w_ii * X[:,ii]
-                    daxpy(n_samples, w_ii,
-                          <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                          1, <DOUBLE*>R.data, 1)
+                    axpy(n_samples, w_ii, &X_data[ii * n_samples], 1,
+                         R_data, 1)
 
                 # tmp = (X[:,ii]*R).sum()
-                tmp = ddot(n_samples,
-                           <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                           1, <DOUBLE*>R.data, 1)
+                if floating is float:
+                    tmp = dsdot(n_samples, &X_data[ii * n_samples], 1, R_data, 1)
```

Comment:
Yes, I've realise. It's fine as is. Sorry I haven't got around to giving this a full review yet, but I'm off to sleep.